flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/hybrid/tiered/file/ProducerMergedPartitionFileReader.java

```diff
@@ -60,9 +141,122 @@ public long getPriority(
             TieredStorageSubpartitionId subpartitionId,
             int segmentId,
             int bufferIndex) {
-        return 0;
+        Tuple2<TieredStorageSubpartitionId, Integer> cacheKey =
+                Tuple2.of(subpartitionId, bufferIndex);
+        RegionCache regionCache = regionCaches.get(cacheKey);
+        return regionCache == null ? Long.MAX_VALUE : regionCache.getFileOffset();
     }
 
     @Override
-    public void release() {}
+    public void release() {
+        if (fileChannel != null) {
+            try {
+                fileChannel.close();
+            } catch (IOException e) {
+                ExceptionUtils.rethrow(e, "Failed to close file channel.");
+            }
+        }
+        IOUtils.deleteFileQuietly(dataFilePath);
+    }
+
+    /**
+     * Initialize the file channel in a lazy manner, which can reduce usage of the file descriptor
+     * resource.
+     *
+     * @param partitionId denotes the partition id.
+     * @throws IOException is thrown in the event of an error.
+     */
+    private void lazyInitializeFileChannel(TieredStoragePartitionId partitionId)
+            throws IOException {
+        if (fileChannel == null) {
+            try {
+                fileChannel = FileChannel.open(dataFilePath, StandardOpenOption.READ);
+            } catch (FileNotFoundException e) {
+                throw new PartitionNotFoundException(
+                        TieredStorageIdMappingUtils.convertId(partitionId));
+            }
+        }
+    }
+
+    /**
+     * Try to get the region cache associated with the given cache key.
+     *
+     * <p>If the relevant region cache exists within the region caches, it will be returned and
+     * subsequently removed. However, if the region cache does not exist, a new region cache will be
+     * created using the data index and returned.
+     *
+     * @param cacheKey denotes the key of the desired region cache.
+     * @return returns the relevant region cache if it exists, otherwise a null value.
+     * @throws IOException is thrown in the event of an error.
+     */
+    private RegionCache tryGetRegionCache(Tuple2<TieredStorageSubpartitionId, Integer> cacheKey)
+            throws IOException {
+        RegionCache regionCache = regionCaches.remove(cacheKey);
+        if (regionCache == null) {
+            Optional<Region> region = dataIndex.getRegion(cacheKey.f0, cacheKey.f1);
+            if (region.isPresent()) {
+                regionCache = new RegionCache(cacheKey.f1, region.get());
+            }
+        } else {
+            numRegionCache--;
+        }
+        return regionCache;
+    }
+
+    /**
+     * The {@link RegionCache} represents the cache of a {@link Region}. Each region cache contains
+     * the last consumed {@link Region}, the next buffer index to consume within the region, and the
+     * file offset of the buffer index.
+     */
+    private class RegionCache {
```

Comment:
Might be better to name this `BufferOffsetCache`

Reply:
Fixed.